consensus/src/dag/dag_handler.rs

```diff
@@ -67,10 +66,35 @@ impl NetworkHandler {
         _buffer: Vec<DAGMessage>,
     ) -> SyncOutcome {
         // TODO: process buffer
+
+        // TODO: Pass bounded executor from EpochManager.
+        let executor = BoundedExecutor::new(2, Handle::current());
+        let epoch_state = self.epoch_state.clone();
+        let mut verified_msg_stream =
+            dag_rpc_rx.concurrent_map(executor, move |rpc_request: IncomingDAGRequest| {
```

Comment:
have you looked at `flat_map_unordered` from futures-util? I had a similar example using `for_each_concurrent` here https://github.com/aptos-labs/aptos-core/blob/601b59774696438f8329b14c378b19ebcf7cd181/network/framework/src/protocols/network/mod.rs#L197

Reply:
Okay, I have simplified the implementation a lot with `flat_map_unordered`. thanks for the feedback!

The downside is that currently I am not able to make this a trait impl. The support for returning `impl Fn` in trait is coming in `v1.75`, and right now, the option is to box pin but it wants the input stream to be 'static and that's not possible.

For now, we need to use `concurrent_map` fn, but post 1.75 upgrade, we can use `stream.concurrent_map`